source-http-ingest/src/server.rs

```diff
@@ -233,35 +241,67 @@ impl Handler {
         &self,
         collection_path: String,
         request_headers: axum::http::header::HeaderMap,
-        body: serde_json::Value,
-    ) -> anyhow::Result<(StatusCode, Json<serde_json::Value>)> {
+        body: Bytes,
+    ) -> anyhow::Result<(StatusCode, Json<Value>)> {
         if let Some(expected_header) = self.require_auth_header.as_ref() {
             match required_string_header(&request_headers, "Authorization") {
                 Ok(auth) if auth == expected_header => { /* request is authorized */ }
                 Ok(_) => {
                     return Ok(err_response(
                         StatusCode::FORBIDDEN,
                         anyhow::anyhow!("invalid authorization token"),
+                        &collection_path,
                     ));
                 }
                 Err(err) => {
-                    return Ok(err_response(StatusCode::UNAUTHORIZED, err));
+                    return Ok(err_response(
+                        StatusCode::UNAUTHORIZED,
+                        err,
+                        &collection_path,
+                    ));
                 }
             }
         }
 
+        // Try to deserialize the body as json _before_ acquiring the handlers lock,
+        // to prevent slow connection from impacting others.
+        let json: Value = match serde_json::from_slice(&body) {
+            Ok(j) => j,
+            Err(err) => {
+                let content_type = request_headers.get("content-type");
+                let content_encoding = request_headers.get("content-encoding");
+
+                let req_body = String::from_utf8_lossy(&body);
```

Comment:
perhaps write a bounded prefix? otherwise this could blow out the log volume pretty easily

Reply:
Good idea. Done.